src/Microsoft.VisualStudio.Threading/AsyncSemaphore.cs

```diff
@@ -149,25 +149,34 @@ private Task<Releaser> LockWaitingHelper(Task waitTask)
         {
             Requires.NotNull(waitTask, nameof(waitTask));
 
-            return waitTask.IsCompleted
```

Comment:
Seems like it could also be addressed by using IsCompletedSuccessfully instead of IsCompleted.  But what you changed it to is fine as well.

Reply:
That's not available in net472 or netstandard2.0 which is what we target today.

Reply:
In that case `waitTask.IsCompleted` => `waitTask.Status == TaskStatus.RanToCompletion`.  Again, what's here is fine, it's just adding a bit more code to optimize the canceled case.

Reply:
And the faulted task still throws the same exception. So that seems like a good way to minimize the diff.

Reply:
PR is now updated using the simpler form 👍